### PR TITLE
T314 remove fillers

### LINKED ISSUE #314: Remove all append_fillers
**Describe the issue**
Currently, by using append.filler, it forces the code to repeat itself. Therefore, i propose that we remove it.

### PR DESCRIPTION
**Pull request by frisograce**
**Please check the following before creating the pull request (PR):**
- [x] Did you run automatic tests?
- [ ] Did you run manual tests?
- [x] Is the code provided in the PR still backwards compatible to previous SIMPA versions?

 Fixes #314

### CODE REVIEW

**Comment by jgroehl** [comment]
Hello!

I'm not quite sure I understand the changes.
I agree that the "filler" argument bloats the interface and implementation but it was a necessary change.
As soon as heterogeneities are introduced, the blood volume fraction per voxel is not strictly set to 1 any more and can be less than 1 because of the random nature of the heterogeneity maps.

In this PR, I do not see an alternative way to compensate for this issue. Perhaps I just missed it...?

**Comment by frisograce** [comment]
Hi!

Sorry, I'm not quite understanding the question. Why would we always want blood volume fractions to be one? As surely everywhere except the blood vessels will have a blood volume less than 1? In regards to total volume, this of course must always add to one I agree. However, one thing I have now seen and agree with is that if the volume fraction given by the user is too big in muscle fraction etc, it will make 

`.append(value=MOLECULE_LIBRARY.muscle_scatterer(volume_fraction=1 - fraction_oxy - fraction_deoxy - water_volume_fraction)`

not work, as the volume fraction will become negative. This will happen regardless of these changes, but perhaps something to consider. An error should be created here to specify to the user what precisely has gone wrong.

It does also speak to how the two methods work. It shows how when removing filler, we end up with the 'excess' volume occupied by the scatterer. With the filler, we end up with a slightly strange occurrence. Here, we specify the custom water volume with a constant again, and then fill the rest with the muscle_scatterer. However, we then go on to add the custom water as a filler, however the only value it can take is still the constant water volume fraction as this will be the only volume left to occupy. Therefore it creates more computation rather unnecessarily.

I hope this might help clarify some of your question, but be sure to let me know if/what I have misunderstood.

**Comment by jgroehl** [comment]
Oh sorry, I chose the wrong words regarding volume fractions and blood volume fractions..
Of course - as soon as one of the fractions is an array, the entire expression is an array that works seamlessly. I think I have just missed that in the early implementation!

File: simpa/utils/libraries/molecule_library.py [hunk 1]
```diff
@@ -71,24 +68,6 @@ def update_internal_properties(self, settings):
             self.internal_properties[Tags.DATA_FIELD_ALPHA_COEFF] += molecule.volume_fraction * \
                 molecule.alpha_coefficient
 
-        if self.filler is not None:
-            filling_volume_fraction = 1.0 - self.internal_properties.volume_fraction
-            if (filling_volume_fraction < 0.0).any():
-                raise AssertionError("The total volume fraction defined was larger than 1.0."
-                                     "Please double check the bounds of the molecular compositions.")
```
Comment (jgroehl) [inline simpa/utils/libraries/molecule_library.py:78]: Apparently we tested "total volume > 1" only if a filler was defined.
It is for sure worth, as you mentioned, to add a check for that!

**Review by jgroehl — APPROVED** [review]

**Comment by frisograce** [comment]
No worries, I completely missed it too! It was only when optimising that i saw it. Will add a check :)

**Pull request merged**